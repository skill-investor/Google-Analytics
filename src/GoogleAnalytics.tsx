import React from "react";
import Script from "next/script";

export function GoogleAnalytics(): JSX.Element | null {
  const gaMeasurementId = process.env.NEXT_PUBLIC_GA_MEASUREMENT_ID;

  if (!gaMeasurementId) {
    return null;
  }

  return (
    <>
      <Script
        src={`https://www.googletagmanager.com/gtag/js?id=${gaMeasurementId}`}
      />
      <Script
        dangerouslySetInnerHTML={{
          __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${gaMeasurementId}', {
              page_path: window.location.pathname,
            });
          `,
        }}
      />
    </>
  );
}
