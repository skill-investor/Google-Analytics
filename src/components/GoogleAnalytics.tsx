import React from "react";
import Script, { ScriptProps } from "next/script";

type GoogleAnalyticsProps = {
  measurementId?: string;
  strategy?: ScriptProps["strategy"];
};

export function GoogleAnalytics({
  measurementId,
  strategy = "afterInteractive",
}: GoogleAnalyticsProps): JSX.Element | null {
  const gaMeasurementId =
    process.env.NEXT_PUBLIC_GA_MEASUREMENT_ID ?? measurementId;

  if (!gaMeasurementId) {
    return null;
  }

  return (
    <>
      <Script
        src={`https://www.googletagmanager.com/gtag/js?id=${gaMeasurementId}`}
        strategy={strategy}
      />
      <Script id="nextjs-google-analytics">
        {`
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '${gaMeasurementId}', {
              page_path: window.location.pathname,
            });
          `}
      </Script>
    </>
  );
}
