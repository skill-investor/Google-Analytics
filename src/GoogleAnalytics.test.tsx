import React from "react";
import renderer from "react-test-renderer";
import { GoogleAnalytics } from "./GoogleAnalytics";

const OLD_ENV = process.env;

beforeEach(() => {
  jest.resetModules();
  process.env = { ...OLD_ENV };
});

afterAll(() => {
  process.env = OLD_ENV;
});

it("renders correctly when measurement id is set", () => {
  process.env.NEXT_PUBLIC_GA_MEASUREMENT_ID = "mock";

  expect(renderer.create(<GoogleAnalytics />).toJSON()).toMatchSnapshot();
});

it("renders null when measurement id is not set", () => {
  process.env.NEXT_PUBLIC_GA_MEASUREMENT_ID = undefined;

  expect(renderer.create(<GoogleAnalytics />).toJSON()).toMatchSnapshot();
});
