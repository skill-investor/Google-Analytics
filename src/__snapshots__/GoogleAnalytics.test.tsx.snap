// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when measurement id is set 1`] = `null`;

exports[`renders null when measurement id is not set 1`] = `null`;
